### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by AntonioHincapie**
- Make sure YAGNI, KISS, and DRY rules and Ruby style guides are respected.
- Follow SOLID principles.
- Pay special attention to the "Single-responsibility Principle".
  - Make sure that the role of the main.RB is only to start your code and ask the user for the initial choice of options (or to quit).
  - Extract some pieces of logic into a separate class(es).

### CODE REVIEW

File: files/app.rb [hunk 1]
```diff
@@ -12,35 +12,6 @@ def initialize
     @rentals = []
   end
```
Comment (brytebee) [inline files/app.rb:13]: - [ ] **Required:** The app crashes when I attempt to run it. See the image below for the stack trace:

image

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @AntonioHincapie👋,

Good job so far coming so far with the progress you've made! 💪💪

## Highlights
You have done amazing in the following:
- Good `README`, well-documented project.✔️
- Good commit messages and PR summaries.✔️
- The app is refactored.✔️
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: files/app.rb (hunk 1 above)
Comment (Davidosky007) [reply files/app.rb:13]: * This error still persists kindly fix it 🙏 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @AntonioHincapie @franclobo 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* Well written PR description 👍 
* Proper use of Gitflow 👍 
* Good looking readme file 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AntonioHincapie** [comment]
> Hi @AntonioHincapie @franclobo 👋,
> 
> Great work on making the changes requested by a previous reviewer 👏 You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
> 
> ## Highlights
> * Well-written PR description 👍
> * Proper use of Gitflow 👍
> * Good looking readme file 👍
> 
> ## Required Changes ♻️
> _Check the comments under the review._
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me @Davidosky007 in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Hi, @Davidosky007
Can you please show me how are you running the app?
As you can see in the image below, the app is working properly on my machine.

image

Please can you clarify this issue for us?
Thanks for your feedback and hope to hear soon from you.
Regards

**Comment by Davidosky007** [comment]
When I try to select options from the terminal I get errors kindly look into it

**Comment by AntonioHincapie** [comment]
Hi @Davidosky007
Yeah, I found the bug.
Please, can you review it and tell me if it's running for you? 
Thanks for your help.

**Review by Davidosky007 — APPROVED** [review]
Hi @AntonioHincapie @franclobo 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* Well written PR description 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**